File: leetcode/Python/0338-m-counting-bits.py
"""
DP
"""
class Solution:
    def countBits(self, num: int) -> List[int]:
        f = [0 for i in range(num + 1)]
        f[0] = 0
        for i in range(1, num+1):
            f[i] = (i & 1) + f[i >> 1]
        return f

File: leetcode/C++/0338-m-counting-bits.cpp
// DP
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> f(num + 1);
        f[0] = 0;
        for (int i = 1; i <= num; i++) {
            f[i] = (i % 2) + f[i >> 1];
        }
        return f;
    }
};